sourcegraph/cody
# custom recipes: add /open command and improve prompts

1. add `/open` command  - the main purpose is to make it easier for users to build a recipe that involves opening a file
1. improve prompts used in custom recipes sample
1. fix JSON format issue
1. cleared up some helper functions

Example recipe that utilizes the /open command:

```json
{
  "title": "Cody Custom Recipes - Workspace",
  "description": "This is an example file to showcase how to build custom recipes for Cody by Sourcegraph.",
  "recipes": {
    "Open User Recipes File": {
      "prompt": "/open user",
      "note": "Open the existing cody.json file for User recipes in your editor if any"
    }
  }
}

```

## Test plan



### To test the /open command

- Enter `/open package.json` in Cody's chatbox with a workspace opened. the package.json file should be opened for you
- Enter `/open user` to open the .vscode/cody.json located in your home directory

### To test the recipe samples

1. sign into Cody with your s2 account
2. copy the recipes from the sample files to the recipes field within your .vscode/cody.json file
3. save the file
4. run the recipe
